src/WebJobs.Extensions.DurableTask/Listener/TaskActorShim.cs

```diff
@@ -0,0 +1,323 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using DurableTask.Core.Common;
+using DurableTask.Core.Exceptions;
+using DurableTask.Core.Middleware;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Implements the actor scheduler as a looping orchestration.
+    /// There is one such orchestration per actor.
+    /// The orchestration terminates if the actor is deleted and idle.
+    /// The orchestration calls ContinueAsNew when it is idle, but not deleted.
+    /// </summary>
+    internal class TaskActorShim : TaskCommonShim
+    {
+        private readonly DurableActorContext context;
+
+        private TaskCompletionSource<object> shutDownSignal
+            = new TaskCompletionSource<object>();
+
+        private Task lastStartedOperation = Task.CompletedTask;
+
+        private Dictionary<Guid, Task> currentReentrantRequests = new Dictionary<Guid, Task>();
+
+        private int remainingReceiveLimit = 20; // TODO make configurable and/or compute based on state size
+
+        public TaskActorShim(DurableTaskExtension config, string schedulerId)
+            : base(config)
+        {
+            this.SchedulerId = schedulerId;
+            this.ActorId = GetActorIdFromSchedulerId(schedulerId);
+            this.context = new DurableActorContext(config, this.ActorId);
+        }
+
+        public override DurableCommonContext Context => context;
+
+        public string SchedulerId { get; private set; }
+
+        public ActorId ActorId { get; private set; }
+
+        public static string GetSchedulerIdFromActorId(ActorId actor)
+        {
+            return $"@{@actor.ActorClass}@{actor.ActorKey}";
+        }
+
+        public static ActorId GetActorIdFromSchedulerId(string schedulerId)
+        {
+            var pos = schedulerId.IndexOf('@', 1);
+            var actorClass = schedulerId.Substring(1, pos - 1);
+            var actorKey = schedulerId.Substring(pos + 1);
+            return new ActorId(actorClass, actorKey);
+        }
+
+        public override RegisteredFunctionInfo GetFunctionInfo()
+        {
+            FunctionName actorFunction = new FunctionName(this.Context.FunctionName);
+            return this.Config.GetActorInfo(actorFunction);
+        }
+
+        public override string GetStatus()
+        {
+            if (this.lastStartedOperation.IsCompleted)
+            {
+                return "Available";
+            }
+            else
+            {
+                if (this.context.State.Queue != null)
+                {
+                    return $"Busy ({this.context.State.Queue.Count} Queued)";
+                }
+                else
+                {
+                    return "Busy";
```

Comment:
What does it mean for an actor to be processing an operation but not have anything queued? Can you add some comments differenciating these cases?

Reply:
Sure, I will add some comments.
"busy" means an operation is currently in progress, e.g. waiting for an activity to complete. You can thus have a busy actor but no other operations waiting in the queue.